src/activation.jl

```diff
@@ -35,6 +44,14 @@ logσ(x::Real) = -softplus(-x)
 const logsigmoid = logσ
 
 
+"""
+    hardtanh(x) = max(-1, min(1, x))
+
+Segment-wise linear approximation of tanh. Cheaper  and  more  computational  efficient version of tanh.
+"""
+hardtanh(x::Real) = oftype(x / 1,max(-1, min( 1, x)))
```

Comment:
```suggestion
hardtanh(x::Real) = max(-one(x), min(one(x), x))
```
I think `hardtanh` doesn't have to force floating point outputs. Our `relu` and `relu6` return integers for integer inputs.

Reply:
Actually I am unsure about how to decide on that. Though without forcing floating point output, all tests are passing(making this as an exception for int to forced floating point test). So, I guess this doesn't have to force floating point outputs. Also, I think the same goes for trelu as well. If that is so, I will make changes in both. Thanks for pointing that out.
Other than unwanted computation cost, are there any other reasons why we ensure such typecastings?

Reply:
I'm not sure the background behind it, but this test 
https://github.com/FluxML/NNlib.jl/blob/c82a76d40cf5c34e4a1eed2655aa30b464e3e408/test/activation.jl#L106-L110

was added in the commit https://github.com/FluxML/NNlib.jl/commit/b698441cf0dfbf08875fa3364b44d95bad4ba104 by @staticfloat .

Reply:
Unless we have a reason to force floating-point (e.g. for gradients) I prefer to keep datatypes as they are.

Reply:
I will the unwanted typecasting then